node/common/src/service/p2p.rs

```diff
@@ -87,4 +87,13 @@ impl P2pCryptoService for NodeService {
             .sign(&msg)
             .expect("unable to create signature")
     }
+
+    fn verify_publication(&mut self, pk: &[u8], publication: &[u8], sig: &[u8]) -> bool {
+        let Ok(pk) = libp2p_identity::PublicKey::try_decode_protobuf(pk) else {
+            return false;
```

Comment:
Is it ok just to return false on decode errors or should we propagate errors?
We might end silently ignoring some errors

Reply:
Yes, you are right. I will add TODO. Higher, we just silently ignore the message, no matter if it is malformed or the signature is invalid. Anyway, we don't have a punishment system for peers who misbehave, so we can't do anything about it.

https://github.com/openmina/openmina/blob/cce612e7edcedffb8af01877b0c3ad7a5351a8c3/p2p/src/network/pubsub/p2p_network_pubsub_effects.rs#L146-L155